scripts/Mansion.gd

```diff
@@ -582,36 +583,88 @@ func _on_end_pressed():
+									person.add_trait('Sex-crazed')
+									hungryforfood = 1.5
+									person.stress += 20
+									person.obed -= max(35 - person.loyal/3,10)
+									person.attention += 40
+									person.attention += 40
+								elif person.metrics.mana_hunger > variables.succubushungerlevel[0] * variables.basemanafoodconsumption * variables.succubusagemod[person.age]:
+									person.lust += 25
+									hungryforfood = 1.5
+									person.stress += 15
+									person.obed -= max(35 - person.loyal/3,10)
+									person.attention += 25
+					#elif person.race_display in ['Golem']:
+					# make the Golem sleep?
+			if hungryforfood > 0: #ralphC - just indents below except where commented #ralphC
+				if chef != null:
+					consumption = max(3, consumption - (chef.sagi + (chef.wit/20))/2)
+					###---Added by Expansion---### Hybrid Support
+					if chef.race.find('Scylla') >= 0:
+						consumption = max(3, consumption - 1)
+					###---End Expansion---###
+				if person.traits.has("Small Eater"):
+					consumption = consumption/3
+				###---Added by Expansion---### ---PENDING: Add option to "Drink from the Source"
+				consumption = consumption * hungryforfood #ralphC - hungryforfood should be 1 unless starving Succubus
+				if person.traits.has("Altered Dietary Needs"): 
+					if globals.expansion.altereddiet_foodavailable(person) == true:
+						temptext
+						consumption = 0
+						temptext = globals.expansion.altereddiet_consumebottle(person)
+						text2.set_bbcode(text2.get_bbcode()+person.dictionary(temptext))
+					else:
+						consumption = consumption * 2
+						text0.set_bbcode(text0.get_bbcode()+person.dictionary('$name could not find any bottles that matched their new dietary needs and had to try to consume twice as much food to avoid starvation. \n'))
 			###---Expansion End---###
-			if globals.resources.food >= consumption:
-				person.loyal += rand_range(0,1)
-				person.obed += person.loyal/5 - (person.cour+person.conf)/10
-				globals.resources.food -= consumption
-			else:
-				person.stress += 20
-				person.health -= rand_range(person.stats.health_max/6,person.stats.health_max/4)
-				person.obed -= max(35 - person.loyal/3,10)
-				if person.health < 1:
-					text = person.dictionary('[color=#ff4949]$name has died of starvation.[/color]\n')
-					deads_array.append({number = count, reason = text})
+				if globals.resources.food >= consumption:
+					person.loyal += rand_range(0,1)
+					person.obed += person.loyal/5 - (person.cour+person.conf)/10
+					globals.resources.food -= consumption
+				else:
+					person.stress += 20
+					person.health -= rand_range(person.stats.health_max/6,person.stats.health_max/4)
+					person.obed -= max(35 - person.loyal/3,10)
+					if person.health < 1:
+						text = person.dictionary('[color=#ff4949]$name has died of starvation.[/color]\n')
+						deads_array.append({number = count, reason = text})
+			#/ralphC
```

Comment:
Loop should `continue` here rather than process the remainder of the loop. This is why toxicity is blamed.

Reply:
comment incorporated for next commit